Library/Usage.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Recurly
+{
+    public class Usage : RecurlyEntity {
+
+        public enum Type
+        {
+            Price,
+            Percentage
+        }
+
+        public int? UnitAmountInCents { get; set; }
+        public float? UsagePercentage { get; set; }
+        public int Amount{ get; set; }
+        public String MerchantTag { get; set; }
+        public Type UsageType { get; set; }
+        public DateTime? UsageTimestamp { get; set; }
```

Comment:
Correct. It is best practice to always send up a usage timestamp (so that we know when the usage is for). The recording timestamp is optional depending on how the merchant integration is set up.